### PR TITLE
Add support for disabling the FIPS mode in OpenJDK

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

When running Strimzi in FIPS enabled Kubernetes or OpenShift clusters. the new versions of OpenJDK automatically enforces the FIPS limitations and allows the applications to use only SunPKCS11 security provider. This is currently not supported by Strimzi and the operator fails (and so would the operands if deployed).

While we currently do not support running Strimzi in FIPS compliant mode, this PR allows to disable the OpenJDK FIPS mode by setting the `FIPS_MODE` environment variable to `disabled`. This will be propagated to all operands and the FIPS mode will be automatically disabled in them.

To implement this, I used the existing mechanism to pass the HTTP Proxy environment variables. This passes environment variables to all operands with minimal code changes. The only changes related to this I did was to:
* Rename the static variable to more generic name sinc eit is not used just for the HTTP Proxy configuration
* Move the environment variable names tot he `ClusterOperatorConfiguration` and refer to them from all places where they were used. They are configured on the CO, so this makes it easier to see the options the CO supports.
* In the shell scripts, where possible, I created a shared module with the methods for handling TLS certificates which is sourced from all the other run scripts. This is also the place where the FIPS mode is disabled for the `keytool` commands.
* The FIPS mode is also passed to the different operands as part of `KAFKA_OPTS` or `JAVA_OPTS` environment variables to which it is added in the startup ecripts.

This PR should cover all operands and operators apart from the Kafka Bridge which is handled in strimzi/strimzi-kafka-bridge#583.

This among other things relates to #6039.

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

File: docker-images/kafka-based/kafka/cruise-control-scripts/cruise_control_tls_prepare_certificates.sh [hunk 1]
```diff
@@ -2,25 +2,10 @@
...
-# $4: Private key to be imported
-# $5: CA public key to be imported
-# $6: Alias of the certificate
-function create_keystore {
-   RANDFILE=/tmp/.rnd openssl pkcs12 -export -in "$3" -inkey "$4" -name "$6" -password pass:"$2" -out "$1"
-}
+# Load predefined functions for preparing trust- and keystores
+# Shellcheck SC1091 is disabled because the path does not work locally, but does inside the container
+# shellcheck disable=SC1091
+source ../kafka/tls_utils.sh
```
Comment (Frawless) [inline docker-images/kafka-based/kafka/cruise-control-scripts/cruise_control_tls_prepare_certificates.sh:8]:
```suggestion
source ../scripts/tls_utils.sh
```

shouldn't be the path be like that?

Comment (scholzj) [reply docker-images/kafka-based/kafka/cruise-control-scripts/cruise_control_tls_prepare_certificates.sh:8]: No, that is not the location in the container images.

**Review by ppatierno — APPROVED** [review]
LGTM. No more than the comment from Jakub.

**Review by Frawless — APPROVED** [review]

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc [hunk 2]
```diff
@@ -0,0 +1,91 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-cluster-operator.adoc
+
+[id='proc-configuring-fips-mode-cluster-operator-{context}']
+= Configuring the FIPS mode in Cluster Operator
+
+[role="_abstract"]
+When running Strimzi on FIPS enabled Kubernetes cluster, the OpenJDK used in Strimzi container images will automatically switch to FIPS mode.
```
Comment (PaulRMellor) [inline documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:9]:
```suggestion
Federal Information Processing Standards (FIPS) are standards for computer security and interoperability.
When running Strimzi on a FIPS-enabled Kubernetes cluster, the OpenJDK used in Strimzi container images will automatically switch to FIPS mode.
```

Comment (PaulRMellor) [reply documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:9]: Does the FIPs security level make a difference?

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+Strimzi currently doesn't support running as FIPS compliant with the FIPS mode enabled.
```
Comment (PaulRMellor) [inline documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:10]:
```suggestion
This prevents Strimzi from running on the cluster.
```

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+When you deploy Strimzi on such cluster, you will see errors similar to this:
```
Comment (PaulRMellor) [inline documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:11]:
```suggestion
When you deploy Strimzi to the cluster, you will see errors similar to the following:
```

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+[source%nowrap]
+----
+Exception in thread "main" io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
+	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:103)
+	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:97)
+	at io.fabric8.kubernetes.client.utils.HttpClientUtils.applyCommonConfiguration(HttpClientUtils.java:210)
```
Comment (PaulRMellor) [inline documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:18]: do we need to show all these lines?

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	... 7 more
+Caused by: sun.security.pkcs11.wrapper.PKCS11Exception: CKR_SESSION_READ_ONLY
+	at jdk.crypto.cryptoki/sun.security.pkcs11.wrapper.PKCS11.C_CreateObject(Native Method)
+	at jdk.crypto.cryptoki/sun.security.pkcs11.wrapper.PKCS11$FIPSPKCS11.C_CreateObject(PKCS11.java:1950)
+	at jdk.crypto.cryptoki/sun.security.pkcs11.P11KeyStore.storeCert(P11KeyStore.java:1567)
+	at jdk.crypto.cryptoki/sun.security.pkcs11.P11KeyStore.engineSetEntry(P11KeyStore.java:1045)
+	... 14 more
+----
+
+If you want to run Strimzi on your FIPS enabled cluster, you can disable the FIPS mode by setting the `FIPS_MODE` environment variable to `disabled`.
```
Comment (PaulRMellor) [inline documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:44]:
```suggestion
If you want to run Strimzi on your FIPS-enabled cluster, you can disable the OpenJDK FIPS mode by setting the `FIPS_MODE` environment variable to `disabled` in the deployment configuration for the Cluster Operator.
```

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+In this mode, the FIPS mode in OpenJDK will be disabled.
```
Comment (PaulRMellor) [inline documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:45]:
```suggestion

```

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+Such Strimzi deployment will not be FIPS compliant, but the Strimzi operators as well as all its operands will be able to run as on any regular Kubernetes cluster.
```
Comment (PaulRMellor) [inline documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:46]:
```suggestion
The Strimzi deployment won't be FIPS compliant, but the Strimzi operators as well as all its operands will be able to run on the FIPS-enabled Kubernetes cluster.
```

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+.Procedure
+
+. To disable the FIPS mode in the Cluster Operator, update its `Deployment` configuration (`install/cluster-operator/060-Deployment-strimzi-cluster-operator.yaml`) and add the `FIPS_MODE` environment variable.
++
+--
+.Example proxy configuration for the Cluster Operator
```
Comment (PaulRMellor) [inline documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:53]:
```suggestion
.Example FIPS configuration for the Cluster Operator
```

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+      serviceAccountName: strimzi-cluster-operator
+      containers:
+        # ...
+        env:
+        # ...
+        - name: "FIPS_MODE"
+          value: "disabled" # <1>
+  # ...
+----
+<1> Disabled the FIPS mode.
```
Comment (PaulRMellor) [inline documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:71]:
```suggestion
<1> Disables the FIPS mode.
```

**Review by PaulRMellor — APPROVED** [review]
Looks good :+1:

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc (hunk 2 above)
Comment (scholzj) [reply documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:9]: What exactly do you mean with FIPS security level? AFAIK on the software level, it is currently just an ON/OFF flag. There are no levels.

File: documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc (hunk 5 above)
Comment (scholzj) [reply documentation/modules/operators/proc-configuring-fips-mode-cluster-operator.adoc:19]: I can try to comment out some of the lines. But the exact exception might change between releases. So I do not think we can leave just the first line. And I'm not sure if it matters that we replace half of the lines with comments. But I can try that

**Pull request merged**